src/ptvsd/_vendored/pydevd/pydevd.py

```diff
@@ -563,12 +587,18 @@ def get_file_type(self, abs_real_path_and_basename):
             None:
                 If it's a regular user file which should be traced.
         '''
-        basename = abs_real_path_and_basename[-1]
-        if basename.startswith('<frozen '):
-            # In Python 3.7 "<frozen ..." appear multiple times during import and should be
-            # ignored for the user.
-            return self.PYDEV_FILE
-        return self._dont_trace_get_file_type(basename)
+        abs_path = abs_real_path_and_basename[0]
```

Comment:
Really minor optimizations (but this is in the critical path performance-wise for the debugger, so, I think it's worth it):

Start the function as:

```
try:
    return _cache_file_type[abs_real_path_and_basename[0]]  #Don't assign abs_path unless we have a KeyError.
except:  # Notice: catch all exception to avoid having to load the `KeyError` name.
    abs_path = abs_real_path_and_basename[0]
```